### ISSUE TITLE
corerun debugger fails on exit with "debugger type: mixed (.net core)"

### ISSUE THREAD

**Issue by sandreenko** [dotnet/runtime#65357]
![image](https://user-images.githubusercontent.com/22642771/153979806-f110766f-b4a0-4de5-81b4-1dea52f3a7f8.png)

Repro: run an app that uses both managed and unmanaged code, for example, C# application or crossgen2, set breakpoints in both managed and unmanaged parts, set debugging mode to "mixed .net core". finish the app and see this error with a high probability. I saw this on VS 2019 and it is still happening with VS 2022.

I guess this scenario is used mostly by runtime developers but would be nice to have it fixed because usually after such errors VS code closes all windows and loses project settings if they were not saved.

In this case, I was running crossgen2 on a test that I and @trylek are working on, so I guess Tomas can repro it as well.

PTAL @mangod9

**Comment by tommcdon** [comment]
Adding @hoyosjs. This looks like a debug build of mscordbi which is asserting.  I believe there is a workaround which involves setting env variables which disables asserts and thus should not crash the process on exit.

**Comment by hoyosjs** [comment]
The workaround has been starting devenv from the command line after setting COMPlus_ContinueOnAssert=1. This is not a solution, but the debugger doesn't get assert testing and these can get stale relatively quickly. This still should be investigated.

**Comment by tommcdon** [comment]
@sandreenko Please let us know if the workaround helped to unblock your workflow.  It's a known issue that asserts in debug-builds of the .NET debugger services binaries can cause the debugger to crash.

**Comment by sandreenko** [comment]
> @sandreenko Please let us know if the workaround helped to unblock your workflow. It's a known issue that asserts in debug-builds of the .NET debugger services binaries can cause the debugger to crash.

Hi Tom, no, I still have the same error and VS crash after it with " COMPlus_ContinueOnAssert=1".

**Comment by hoyosjs** [comment]
You start devenv from a cmd with that env var set?

**Comment by sandreenko** [comment]
> You start devenv from a cmd with that env var set?

Oh, forgot that part, thanks @hoyosjs, not it does not assert/fail!

**Comment by tommcdon** [comment]
@sandreenko Thanks for confirming the workaround!
